web-app/js/HighDimensionalData.js

```diff
@@ -394,16 +396,19 @@ HighDimensionalData.prototype.gather_high_dimensional_data = function (divId, hi
       this.fetchNodeDetails( divId, function( result ) {
         _this.data = JSON.parse(result.responseText);
 
-        _this.display_high_dimensional_popup();
-//        TODO: re-enable platform validation, except for geneprint:
-//        platforms = _this.getPlatformValidator(_this.getPlatforms(_this.data));
-//        var formValidator = new FormValidator(platforms);
-//
-//        if (formValidator.validateInputForm()) {
-//          _this.display_high_dimensional_popup();
-//        } else {
-//          formValidator.display_errors();
-//        }
+        if (doValidatePlatforms) {
+          platforms = _this.getPlatformValidator(_this.getPlatforms(_this.data));
+          var formValidator = new FormValidator(platforms);
+
+          if (formValidator.validateInputForm()) {
+            _this.display_high_dimensional_popup();
+          } else {
+            formValidator.display_errors();
+          }
+        }
+        else {
+          _this.display_high_dimensional_popup();
```

Comment:
I'm not sure what you mean. It's only called once, right? Once if validation succeeds or once if no validation is required.